### PR TITLE
build postgres on alpine

### PR DESCRIPTION
**Pull request by davejrt**
This PR builds postgres 12.6 on alpine. 

It has been brought to our attention recently that our debian based image contains multiple upstream vulnerabilities that we cannot patch. In an effort to remediate these, we are building an alpine based image. 

This build eliminates the two binary approach previously contained in our images, where we automatically migrated. By doing so, we reduce our image size, and attack surface for any future vulnerabilities in older versions of postgres. 

Any further migrations can be done out of band by a one off init container. 

Snyk scan of an image build from this PR is [here](https://app.snyk.io/org/dave-b1y/project/d1f20277-18cd-4722-8643-f0e4f5802e9f)

Fixes https://github.com/sourcegraph/security-issues/issues/151

### CODE REVIEW

**Review by ggilmore — APPROVED** [review]

**Comment by davejrt** [comment]
@efritz @sourcegraph/core-application would love to get your input here and talk about any concerns you might have. This has been manually tested on compose/k8s clusters and works without issues but we're happy to talk about anything we may have failed to consider here.

**Comment by davejrt** [comment]
Merging without review from codeintel as eric is on leave. I want to get this into the hands of customers to at least scan and will follow up with him when he's back from leave.

**Comment by efritz** [comment]
My only concerns would be with the startup/migration, but if you've sufficiently tested these on a real cluster I have no blocking (or nit) comments.

File: docker-images/postgres-12.6-alpine/rootfs/initdb.sh [hunk 1]
```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env bash
```
Comment (daxmc99) [inline docker-images/postgres-12.6-alpine/rootfs/initdb.sh:1]: Does our Alpine image contain bash?

**Pull request merged**